jetstream/analysis.py

```diff
@@ -315,15 +324,43 @@ def validate(self) -> None:
         for v in self.config.metrics.values():
             metrics |= {m.metric for m in v}
 
-        sql = exp.build_query_template(
-            metrics,
+        enrollments_sql = exp.build_enrollments_query(
             limits,
             "normandy",
             self.config.experiment.enrollment_query,
             self.config.experiment.segments,
-        ).format(results_table="results_table")
+        )
+
+        dry_run_query(enrollments_sql)
+
+        metrics_sql = exp.build_metrics_query(
+            metrics,
+            limits,
+            "enrollments_table",
+        )
+
+        # enrollments_table doesn't get created when performing a dry run;
+        # the metrics SQL is modified to include a subquery for a mock enrollments_table
+        # A UNION ALL is required here otherwise the dry run fails with
+        # "cannot query over table without filter over columns"
+        metrics_sql = metrics_sql.replace(
+            "WITH enrollments AS (",
+            """WITH enrollments_table AS (
+                SELECT '00000' AS client_id,
+                    'test' AS branch,
+                    DATE('2020-01-01') AS enrollment_date,
+                    1 AS analysis_window_start,
+                    2 AS analysis_window_end
+                UNION ALL
+                SELECT '00000' AS client_id,
+                    'test' AS branch,
+                    DATE('2020-01-01') AS enrollment_date,
+                    1 AS analysis_window_start,
+                    2 AS analysis_window_end
+            ), enrollments AS (""",
+        )
```

Comment:
Maybe not the most elegant thing to do, but it works and doesn't increase query complexity much. I'm open to better suggestions though.

Reply:
I guess the alternative is to publish an enrollment fixture.

I suppose https://github.com/mozilla/jetstream/blob/main/script/jetstream-dryrun.js doesn't have access to do that. We could do something like add a Jetstream command to publish a validation fixture that runs in CI after a deploy.

The search/replace method seems fragile but we control both halves and I think we're unlikely to recover the costs of doing something more sophisticated so let's roll with it.

Reply:
Publishing a fixture would be a possibility, though it would add more complexity to the dry running process since two commands need to be executed and also permissions to write data would need to be granted to the CI service account.

If this search/replace method here causes any problems we can look into the fixture solution, but for now it seems the easiest to implement.